common/src/main/java/com/microsoft/identity/common/adal/internal/cache/ADALOAuth2TokenCache.java

```diff
@@ -393,7 +393,7 @@ private void validateSecretKeySetting() {
         Logger.verbose(TAG, "Validating secret key settings.");
         final byte[] secretKeyData = AuthenticationSettings.INSTANCE.getSecretKeyData();
 
-        if (secretKeyData == null && Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2) {
+        if (secretKeyData == null) {
             throw new IllegalArgumentException("Secret key must be provided for API < 18. "
                     + "Use AuthenticationSettings.INSTANCE.setSecretKey()");
         }
```

Comment:
should this be removed instead, this is opening up throwing exception for >=  JELLY_BEAN_MR2

Reply:
25bad5720fe52f15d9d9bf6de5710f5012cf4713